### ISSUE TITLE
oc cluster up fails for v3.9.0, works for v3.9.0-alpha.3

### ISSUE THREAD

**Issue by pilhuhn** [openshift/origin#19034]
oc cluster up fails for v3.9.0 while it works for v3.9.0-alpha3

##### Version

[root@pintsize ~]# oc version
oc v3.9.0-alpha.3+78ddc10
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

on RHEL 7.4

##### Steps To Reproduce
1. [step 1]
2. [step 2]

[root@pintsize ~]# oc cluster up  --service-catalog=false --public-hostname=172.31.7.9 --version=v3.9.0
Starting OpenShift using openshift/origin:v3.9.0 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v3.9.0 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ...
   WARNING: Binding DNS on port 8053 instead of 53, which may not be resolvable from all clients.
-- Checking type of volume mount ...
   Using nsenter mounter for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ...
   Using public hostname IP 172.31.7.9 as the host IP
   Using 172.31.7.9 as the server IP
-- Starting OpenShift container ...
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Adding default OAuthClient redirect URIs ... OK
-- Installing registry ...
   scc "privileged" added to: ["system:serviceaccount:default:registry"]
-- Installing router ... OK
-- Importing image streams ... OK
-- Importing templates ... OK
-- Importing internal templates ... OK
-- Installing web console ... FAIL
   Error: failed to start the web console server: timed out waiting for the condition

##### Current Result

   Error: failed to start the web console server: timed out waiting for the condition

[root@pintsize ~]# oc project openshift-web-console
Now using project "openshift-web-console" on server "https://172.31.7.9:8443".
[root@pintsize ~]# oc get pods
NAME                         READY     STATUS             RESTARTS   AGE
webconsole-884bc55f7-ltmd8   0/1       CrashLoopBackOff   41         3h

oc logs webconsole-884bc55f7-ltmd8
W0320 15:14:27.214944       1 start.go:93] Warning: config.clusterInfo.loggingPublicURL: Invalid value: "": required to view aggregated container logs in the console, web console start will continue.
W0320 15:14:27.257953       1 start.go:93] Warning: config.clusterInfo.metricsPublicURL: Invalid value: "": required to view cluster metrics in the console, web console start will continue.
Error: AssetConfig.webconsole.config.openshift.io "" is invalid: [config.clusterInfo.consolePublicURL: Invalid value: "": must contain a scheme (e.g. https://), config.clusterInfo.consolePublicURL: Invalid value: "": must contain a host, config.clusterInfo.consolePublicURL: Invalid value: "": must have a trailing slash in path, config.clusterInfo.masterPublicURL: Invalid value: "": must contain a scheme (e.g. https://), config.clusterInfo.masterPublicURL: Invalid value: "": must contain a host]
Usage:
  origin-web-console [flags]
Flags:
[...]
F0320 15:14:27.259271       1 console.go:35] AssetConfig.webconsole.config.openshift.io "" is invalid: [config.clusterInfo.consolePublicURL: Invalid value: "": must contain a scheme (e.g. https://), config.clusterInfo.consolePublicURL: Invalid value: "": must contain a host, config.clusterInfo.consolePublicURL: Invalid value: "": must have a trailing slash in path, config.clusterInfo.masterPublicURL: Invalid value: "": must contain a scheme (e.g. https://), config.clusterInfo.masterPublicURL: Invalid value: "": must contain a host]

##### Expected Result

It works as in v3.9.0-alpha3:

[root@pintsize ~]# oc cluster up  --service-catalog=false --public-hostname=172.31.7.9
Starting OpenShift using openshift/origin:v3.9.0-alpha.3 ...
OpenShift server started.

The server is accessible via web console at:
    https://172.31.7.9:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

[root@pintsize ~]# oc cluster down

##### Additional Information
[try to run `$ oc adm diagnostics` (or `oadm diagnostics`) command if possible]
[if you are reporting issue related to builds, provide build logs with `BUILD_LOGLEVEL=5`]
[consider attaching output of the `$ oc get all -o json -n <namespace>` command to the issue]
[visit https://docs.openshift.org/latest/welcome/index.html]

**Comment by jwforres** [comment]
@deads2k is this because you blew something up in cluster up? console not getting some of its necessary config (consolePublicURL and masterPublicURL)

cc @spadgett

**Comment by jwforres** [comment]
@openshift/sig-master @openshift/sig-user-interface

**Comment by deads2k** [comment]
> @deads2k is this because you blew something up in cluster up? console not getting some of its necessary config (consolePublicURL and masterPublicURL)

My changes merged after we branched, not before.

**Comment by spadgett** [comment]
This is happening because you're using the `v3.9.0-alpha.3+78ddc10` version of oc, which is old. There were API incompatible changes to the web console config since that oc version. You'll need oc built from the current release-3.9 branch to use the current v3.9 images. Unfortunately there is no download yet.

**Comment by spadgett** [comment]
The 3.9 release is coming soon. When available, the 3.9 release version of oc will work correctly with v3.9 images.

**Comment by spadgett** [comment]
This is fixed in the 3.9 release oc. The download is here:
 https://github.com/openshift/origin/releases/tag/v3.9.0

```
~/Downloads/openshift-origin-client-tools-v3.9.0-191fece-mac
❯ ./oc cluster up
Using Docker shared volumes for OpenShift volumes
Using 127.0.0.1 as the server IP
Starting OpenShift using openshift/origin:v3.9.0 ...
OpenShift server started.

The server is accessible via web console at:
    https://127.0.0.1:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin


~/Downloads/openshift-origin-client-tools-v3.9.0-191fece-mac 43s
❯ ./oc version
oc v3.9.0+191fece
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth

Server https://127.0.0.1:8443
openshift v3.9.0+191fece
kubernetes v1.9.1+a0ce1bc657
```

/close
